pkgs/os-specific/linux/r8125/default.nix

```diff
@@ -31,6 +31,9 @@ stdenv.mkDerivation rec {
 
   buildFlags = [ "modules" ];
 
+  # r8125 has been integrated into the kernel as of v5.9.1
+  broken = lib.versionAtLeast kernel.version "5.9.1";
```

Comment:
I think instead of broken you should use `disable` here instead :thinking: ? Not sure.

Reply:
That's probably more appropriate; I didn't know it existed. (is it `disable` or `disabled`?)

Although neither `broken` or `disabled` seem to prevent me from building the derivation:
<details>
<summary>$ nix-build . -A linuxPackages_5_10.r8125 --check</summary>

```
# PWD is my local clone of the repo
$ nix-build . -A linuxPackages_5_10.r8125 --check
checking outputs of '/nix/store/3kdr3p8yv4nfxbccyxpyhj0gcjdyvq5q-r8125-9.004.01.drv'...
unpacking sources
unpacking source archive /nix/store/31fkm175qilc3c0mgx7zxi81464kwcp2-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/4l7wsi6h6283194r6fqy1731qxlagq62-bash-4.4-p23/bin/bash BASEDIR=/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0 modules
make -C src/ modules
make[1]: Entering directory '/build/source/src'
find: '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/kernel/drivers/net/ethernet': No such file or directory
find: '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/kernel/drivers/net': No such file or directory
make -C /nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build M=/build/source/src modules
make[2]: Entering directory '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build'
  CC [M]  /build/source/src/r8125_n.o
  CC [M]  /build/source/src/rtl_eeprom.o
  CC [M]  /build/source/src/rtltool.o
  LD [M]  /build/source/src/r8125.o
  MODPOST /build/source/src/Module.symvers
  CC [M]  /build/source/src/r8125.mod.o
  LD [M]  /build/source/src/r8125.ko
make[2]: Leaving directory '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build'
make[1]: Leaving directory '/build/source/src'
installing
install flags: SHELL=/nix/store/4l7wsi6h6283194r6fqy1731qxlagq62-bash-4.4-p23/bin/bash BASEDIR=/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0 install
make -C src/ install
make[1]: Entering directory '/build/source/src'
find: '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/kernel/drivers/net/ethernet': No such file or directory
find: '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/kernel/drivers/net': No such file or directory
make -C /nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build M=/build/source/src INSTALL_MOD_DIR=kernel/drivers/net INSTALL_MOD_PATH=/nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01 modules_install
make[2]: Entering directory '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build'
  INSTALL /build/source/src/r8125.ko
At main.c:160:
- SSL error:02001002:system library:fopen:No such file or directory: crypto/bio/bss_file.c:69
- SSL error:2006D080:BIO routines:BIO_new_file:no such file: crypto/bio/bss_file.c:76
sign-file: certs/signing_key.pem: No such file or directory
  DEPMOD  5.10.0
Warning: modules_install: missing 'System.map' file. Skipping depmod.
make[2]: Leaving directory '/nix/store/kkpmgr0dmjiha96g94468jvshg7ximi4-linux-5.10-dev/lib/modules/5.10.0/build'
make[1]: Leaving directory '/build/source/src'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01
strip is /nix/store/kcp0y7g1ir4dxq2gqz687i4vb7gqy8s0-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01/lib
strip:/nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01/lib/modules/5.10.0/kernel/drivers/net/r8125.ko.xz: file format not recognized
patching script interpreter paths in /nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01
checking for references to /build/ in /nix/store/77fv77bj6rk6swlibmn74qd4rp6w2w4w-r8125-9.004.01...
```
</details>

Reply:
Oh `broken` is a meta attribute. `disabled` seems to be mostly used in python-related derivations (in the derivation itself, not in the meta attribute).
Seems like `broken` is the only way to prevent a derivation from evaluating/building.